### ISSUE TITLE
Invalid Cast in linux target

### ISSUE THREAD

**Issue by joekinley** [underscorediscovery/luxe#216]
When I use ParcelProgress and try to compile and run, it crashes with an "Invalid cast" crash:

Here is the output: 

    Called from PlayState::init::PlayState.hx::24
    Called from luxe.ParcelProgress::new::luxe/ParcelProgress.hx::94
    Called from luxe.Sprite::new::luxe/Sprite.hx::48
    Called from luxe.Visual::new::luxe/Visual.hx::109
    Called from luxe.Visual::_create_geometry::luxe/Visual.hx::184
    Called from luxe.Sprite::set_geometry::luxe/Sprite.hx::93
    Error : Invalid Cast

I also get this in the sample "/samples/guides/3_sprite_animation". Doesn't happen in web target.

**Comment by underscorediscovery** [comment]
Thanks for the report, will have a look soon.

**Comment by desttinghim** [comment]
I am also having this problem, but on windows.

**Comment by underscorediscovery** [comment]
Found that this is related to the latest hxcpp version, see this thread for details:
https://groups.google.com/forum/#!topic/haxelang/bkDM9Na8JSM

The progress border was using the wrong geometry type for `Sprite` but the previous (more consistent) behaviour was to return null on a failed cast like this, which was ok in this case as it was a pass through state. I have changed this to use `Visual`, as it should, but will have to see what the final decision on hxcpp side is for clarity on the potentially breaking changes on the language semantics. hopefully none.
